tests/capture/test_nested_plxpr.py

```diff
@@ -186,6 +186,51 @@ def workflow(x):
         out = jax.core.eval_jaxpr(plxpr.jaxpr, plxpr.consts, 0.5)
         assert qml.math.isclose(out, qml.math.sin(-(0.5 + 0.3)))
 
+    def test_dynamic_shape_input(self):
+        """Test that the adjoint transform can accept arrays with dynamic shapes."""
+        jax.config.update("jax_dynamic_shapes", True)
+        try:
+
+            def f(x):
+                qml.adjoint(qml.RX)(x, 0)
+
+            jaxpr = jax.make_jaxpr(f, abstracted_axes=("a",))(jax.numpy.arange(4))
+
+            tape = qml.tape.plxpr_to_tape(jaxpr.jaxpr, jaxpr.consts, 2, jax.numpy.arange(2))
+            expected = qml.adjoint(qml.RX(jax.numpy.arange(2), 0))
+            qml.assert_equal(tape[0], expected)
+
+        finally:
+            jax.config.update("jax_dynamic_shapes", False)
+
+    def test_complicated_dynamic_shape_input(self):
+        """Test a dynamic shape input with a more complicate shape."""
+        jax.config.update("jax_dynamic_shapes", True)
+        try:
+
+            def g(x, y):
+                qml.RX(x["a"], 0)
+                qml.RY(y, 0)
+
+            def f(x, y):
+                qml.adjoint(g)(x, y)
+
+            abstracted_axes = ({"a": {0: "n"}}, {0: "m"})
+            jaxpr = jax.make_jaxpr(f, abstracted_axes=abstracted_axes)(
+                {"a": jax.numpy.arange(2)}, jax.numpy.arange(3)
+            )
+            tape = qml.tape.plxpr_to_tape(
+                jaxpr.jaxpr, jaxpr.consts, 3, 4, jax.numpy.arange(3), jax.numpy.arange(4)
+            )
+
+            op1 = qml.adjoint(qml.RY(jax.numpy.arange(4), 0))
+            op2 = qml.adjoint(qml.RX(jax.numpy.arange(3), 0))
+            qml.assert_equal(op1, tape[0])
+            qml.assert_equal(op2, tape[1])
+
+        finally:
+            jax.config.update("jax_dynamic_shapes", False)
```

Comment:
This confused me a lot 😅 

Side note, considering the fact that this file has tests that also enable/disable dynamic shapes, it's probably best to put the fixture in `conftest.py`.

Reply:
Which part is the most confusing? The `abstracted_axes` shapes?

Reply:
The test is fine 😅 . I should've been clearer. The nested structure of the arguments made it a bit hard to parse, but that is the intention, so it's fine.

Reply:
I'll add in a few more explanatory variables.